x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankFeaturePhaseRankCoordinatorContext.java

```diff
@@ -78,6 +79,12 @@ protected void computeScores(RankFeatureDoc[] featureDocs, ActionListener<float[
             }
         });
 
+        // Short circuit on empty results after request validation
+        if (featureDocs.length == 0) {
+            inferenceListener.onResponse(new InferenceAction.Response(new RankedDocsResults(List.of())));
+            return;
+        }
```

Comment:
The tests are failing because this called before the inference ID is validated. I think we can fix it by putting it in `topNListener`, just before `featureData` is created.

Reply:
Thanks, getting back to this today, appreciate the pointer :)